platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -383,6 +390,18 @@ private void logout() {
 
         SecurityTokenHolder tokenHolder = ((SecurityTokenHolder) session.getAttribute(
                 SecurityConstants.SAML_ASSERTION));
+
+        for (SecurityToken token : tokenHolder.getRealmTokenMap()
+                .values()) {
+            try {
+                Subject subject = securityManager.getSubject(token);
+                if (subject.hasRole("admin")) {
```

Comment:
Rather than just checking for "admin", we most likely need to check for the entire range of karaf defined roles: `group,admin,manager,viewer,webconsole,system-admin,system-history,systembundles`. These can also change as we update versions, so we should probably be tossing this list into system.properties so that we don't need to change code in the future.